Upgrade vagrant box to run on python 3.6

Tomasz-Kluczkowski opened the pull request:
Changed vagrant box to bento/ubuntu-18.04 which gives Python 3.6.5.

Fixes #1333

Tomasz-Kluczkowski committed:
Initial changes for upgrading to ubuntu 18.04 / python 3.6.

Tomasz-Kluczkowski committed:
Changed vagrant box to bento/ubuntu-18.04.

Removed change to sources.list.
Removed lxml dependency from pythondotorg.yml.
Postgres version requirement removed.
Changed python version comment in install.rst.
Fixes #1333.

berkerpeksag commented on provisioning/pythondotorg.yml:
```diff
@@ -36,13 +31,6 @@
 
-    - name: Install python-lxml deps
-      become: yes
-      apt: pkg=python-lxml state=build-dep
```
Does this get installed by BeautifulSoup now?

Tomasz-Kluczkowski replied:
Sure I will remove # - acl in the next amendment.
Regading lxml:
I have done local tests yesterday during the Python Sprint of the new box and was able to launch the site with initial data.
Also built the docs from txt to html and run the management command to do the same.
It looks like lxml required works now without specifying those dependencies.
I will retest all of that though just to make sure. Is there anything else that lxml is used for?

berkerpeksag replied:
> Is there anything else that lxml is used for?

I think BeautilfulSoup + lxml is only used by the PEP converter, so running `./manage.py test peps` may help to confirm this.

Gautier replied:
`lxml` is now available as a wheel, so `pip install` does not need to build it any-more which means we don't need the `pkg=python-lxml state=build-dep` step and we can also remove `libxml2-dev` and `libxslt-dev` from the `Install base packages` step.

I have tested locally, `./manage.py test peps` passes and I can generate the PEPs.

berkerpeksag replied:
Awesome, thanks for testing @Gautier! @Tomasz-Kluczkowski could you remove those packages as well?

Tomasz-Kluczkowski replied:
Working on it.

Tomasz-Kluczkowski replied:
Added Gautier to the commit as a co-autor - he helped a lot during the sprint.

berkerpeksag reviewed (commented):
This looks good, but I haven't tested it locally yet. Could you also remove the commented-out line at line 29:

```yml
# - acl
```

berkerpeksag commented on docs/source/install.rst:
```diff
@@ -18,7 +18,7 @@ You should then be able to provision the Vagrant box.
     $ vagrant up
 
-The box will be provisioned with Python 3.5, a virtual environment with all
+The box will be provisioned with Python 3.6.5, a virtual environment with all
```
Nit: How about saying something like "the latest Python 3.6 release" or "version" so we don't need to update the version information every six month :)

Tomasz-Kluczkowski replied:
Ok, will do.
I need a day or two though as was really busy today - work / training :).

berkerpeksag replied:
Sure, there is no rush :)

Tomasz-Kluczkowski committed:
Remove not needed packages and clean up upgrading to bento/ubuntu-18.04.

Removed libxml2-dev, libxslt-dev from packages installed when running
vagrant up.
Removed # - acl comment.
Changed python version description to latest python 3.6 version in
install.rst.

Cloned peps repo, run make -j to build html.
Run vagrant box and run .manage.py generate_pep_pages - all ok,
 peps content accessible once server run.
To be sure I have done the same not in vagrant - on local development
and again all worked.
All tests pass in both environments.

Co-authored-by: Gautier <ghayoun@gmail.com>
